hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/LegacyReplicationManager.java

```diff
@@ -781,14 +645,10 @@ public CompletableFuture<MoveResult> move(ContainerID cid,
    * @param cid Container to move
    * @param mp MoveDataNodePair which contains source and target datanodes
    */
-  public CompletableFuture<MoveResult> move(ContainerID cid,
+  private CompletableFuture<MoveResult> move(ContainerID cid,
       MoveDataNodePair mp)
       throws ContainerNotFoundException, NodeNotFoundException {
     CompletableFuture<MoveResult> ret = new CompletableFuture<>();
-    if (!isRunning()) {
```

Comment:
Is the balancer going to keep working OK if this RM thread is not running any more? From this check, it must have needed / wanted it to be running before?

Reply:
since the legacy RM does not have any scheduling logic, it is just a container processor now and does not know whether the new RM is running. the check is needed, and has been moved to the new RM , please see 
[the check in new RM](https://github.com/apache/ozone/pull/3352/files/dcb2ce3c53bc01ed30aab1b27e494cddfda5a44a#diff-8be983672d581b5fdec0b1a7ce4fe4f0ee29a25684f2af3680604d51d3ebbb52R412)